### DISCUSSION TITLE
Disable local login

### DISCUSSION THREAD

**Discussion by nicedevil007** [TandoorRecipes/recipes#2336] [Q&A] [answered]
Heya, I just enabled OpenIDConnect with authentik and now I linked my existing accounts to my authentik account.
Is is possible to deactivate the local login that doesn't have 2FA f.e. and only allow the authentication via authentik?

**Comment by vabene1111** [comment, marked as answer]
You can disable registration but not login at the moment

**Reply by nicedevil007** [reply]
okay so I have already done everything, nothing more I can do right now to make it more secure. I hope it will be there in the future but thank you anyway!

**Reply by vabene1111** [reply]
its a good idea that i have never tought about so might add it in the future. Wanted to add MFA to tandoor itself anyway at some point but then in the end its only cooking recipes so I have mainly focused on features for now since my time is sadly limited.

**Reply by nicedevil007** [reply]
no problem at all :)
same on my side, not really neccesary for cooking recipes but I try to add everything to SSO now where I can define precisly where a 2FA is needed and where not. 

First step is done, but it doesn't make sense at the end if we are still able to connect to it without the forwarding from our SSO services (like authentik).

**Reply by BobWs** [reply]
@nicedevil007 Hi, would you mind sharing how you did manage to setup SSO with Tandoor/recipes?
I’m also looking to some options to add SSO to some of my docker apps which doesn’t have 2FA at the moment.
And as this is only for recipes it is a great way to test SSO without messing up important apps 😁

I was thinking of using Authentik or Keycloak. 
TIA

**Reply by nicedevil007** [reply]
sure, for myself the linking of existing accounts with OIDC accounts is not working, registering new ones is working well

right now I'm running tandoor in a LXC container and not in docker anymore. there is a `*.env` file that got 3 lines added:

```python
SOCIAL_PROVIDERS=allauth.socialaccount.providers.openid_connect
SOCIALACCOUNT_PROVIDERS={"openid_connect":{"APPS":[{"provider_id":"authentik","name":"Authentik","client_id":"<your client ID>","secret":"<your secret>","settings":{"server_url":"https://<AUTHENIK URL>/application/o/tandoor/.well-known/openid-configuration"}}]}}
```

**Reply by nicedevil007** [reply]
hey @vabene1111 

since 2023 there is some time gone. now I was able to setup OIDC on my paperless instance and paperless is working with django as well. so I guess the implementation of the option is close the same to your tandoor?

```python
PAPERLESS_DISABLE_REGULAR_LOGIN=true
```

https://docs.paperless-ngx.com/configuration/#PAPERLESS_DISABLE_REGULAR_LOGIN

**Reply by BobWs** [reply]
Thank for sharing! If I may ask which SSO are you using? Authentik, Authelia or Keycloak?
I tried Keycloak but it didn't work for me, I coudn't get it to work on my Synology with Nginx-Proxy-Manager...even after reading and watching a lot of tutorials:-(

I will give Authentik I try...I also started using Paperless-ngx recently, so it good to hear you have managed to get it to work with SSO/2FA I was looking for that also;-)
Thanks again!

**Reply by nicedevil007** [reply]
Give authentik a try, works like a charm for most things on my end :)

**Reply by hugofollic** [reply]
Definitely a great feature that I would like to have. Is it complicated to implement it ?

**Reply by tristan-k** [reply]
> You can disable registration but not login at the moment

How do I disable registration?

**Reply by smilerz** [reply]
https://docs.tandoor.dev/system/configuration/#enable-signup

**Comment by sargunv** [comment]
For anyone else who stumbles upon this thread from Google: I was able to hack it for now in my Nginx config:

```nginx
location /api-token-auth {
  return 404;
}

location /accounts/login/ {
    return 303 /accounts/oidc/authentik/login/?process=login;
}
```

The first block disables the login API, and the second automatically redirects from the frontend to OIDC login. Update that latter path for your setup (change authentik to whatever your oidc provider is called).

**Comment by mwllgr** [comment]
Also, to auto-click the button, I overwrote my login.html with the following:

```html
{% extends "base.html" %}
{% load crispy_forms_filters %}
{% load i18n %}

{% block head_title %}{% trans "Signup" %}{% endblock %}

{% block content %}

<h1>Redirecting to login...</h1>
<p>Please wait or click the button below.</p>

<form method="post">
  {% csrf_token %}
  <button type="submit" class="btn btn-success">Not working? Click here.</button>
</form>

<script>
document.getElementsByClassName('btn btn-success')[0].click();
</script>

{% endblock %}
```

... and put it in my docker-compose.yml:
```yaml
[...]
    volumes:
      - ./login.html:/opt/recipes/cookbook/templates/socialaccount/login.html
[...]
```
